pennylane/ops/op_math/composite.py

```diff
@@ -131,14 +134,67 @@ def is_hermitian(self):
     def has_matrix(self):
         return all(op.has_matrix for op in self)
 
-    @abc.abstractmethod
     def eigvals(self):
-        """Return the eigenvalues of the specified operator."""
+        """Return the eigenvalues of the specified operator.
+
+        This method uses pre-stored eigenvalues for standard observables where
+        possible and stores the corresponding eigenvectors from the eigendecomposition.
+
+        Returns:
+            array: array containing the eigenvalues of the operator
+        """
+        eigvals = []
+        for ops in self.overlapping_ops:
+            if len(ops) == 1:
+                eigvals.append(
+                    qml.utils.expand_vector(ops[0].eigvals(), list(ops[0].wires), list(self.wires))
+                )
+            else:
+                tmp_composite = self.__class__(*ops)
```

Comment:
While this looks cleaner, I don't think its very efficient to instantiate a new instance of the class just to access the eigen decomposition method and the wires. We also don't gain any performance boosts from the caching of the Eigenvals since the instance only exists in the scope off the function. 

I think in this case you could just copy some of the logic from eigen decomposition into a helper function and call it directly.

Reply:
The eigendecomposition depends on the matrix of the operators, and the matrix depends on the type of composite operator that you are using. Doing this would require adding a helper function call for each operator that computes the matrix and then the eigenvalues. My question is: is instantiating a new class really that inefficient to implement all these changes?

Reply:
Ahh good point, probably not significant 👍🏼